test/extended/networking/commatrix.go

```diff
@@ -0,0 +1,252 @@
+
+		err := os.MkdirAll(artifactsDir, 0755)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		cs, err := client.New()
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		configClient := configv1client.NewForConfigOrDie(cs.Config)
+
+		deployment := types.Standard
+		isSNO, err := isSNOCluster(configClient)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		if isSNO {
+			deployment = types.SNO
+		}
+
+		env := types.Cloud
+		isBM, err := isBMCluster(configClient)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		if isBM {
+			env = types.Baremetal
+		}
+
+		g.By("get cluster's version and check if it's suitable for test")
+		clusterXYVersion, err := getClusterXYersion(configClient)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		floatClusterXYVersion, err := strconv.ParseFloat(clusterXYVersion, 64)
+		o.Expect(err).ToNot(o.HaveOccurred())
+		if floatClusterXYVersion < minimalDocCommatrixVersion {
+			g.Skip(fmt.Sprintf("If the cluster version is lower than the lowest version that "+
+				"has a documented communication matrix (%v), skip test", minimalDocCommatrixVersion))
+		}
+		docCommatrixVersionedUrl := strings.Replace(docCommatrixBaseUrl, "VERSION", clusterXYVersion, 1)
+		resp, err := http.Get(docCommatrixVersionedUrl)
+		o.Expect(err).ToNot(o.HaveOccurred())
+		defer resp.Body.Close()
+		// expect response status code to differ from status not found
+		docURLStatusIsNotFound := resp.StatusCode == http.StatusNotFound
+		o.Expect(docURLStatusIsNotFound).To(o.BeFalse())
+
+		g.By("preparing for commatrices generation")
+		epExporter, err := endpointslices.New(cs)
+		o.Expect(err).ToNot(o.HaveOccurred())
+		utilsHelpers := utils.New(cs)
+
+		g.By("generating new commatrix")
+		newComMatrixCreator, err := commatrixcreator.New(epExporter, "", "", env, deployment)
+		o.Expect(err).ToNot(o.HaveOccurred())
+		newComMatrix, err := newComMatrixCreator.CreateEndpointMatrix()
+		o.Expect(err).ToNot(o.HaveOccurred())
+		newComMatrix.WriteMatrixToFileByType(utilsHelpers, "new-commatrix", types.FormatCSV, deployment, artifactsDir)
+
+		g.By("get documented commatrix")
+		fp := filepath.Join(artifactsDir, "doc-commatrix.csv")
+		createCSVFromUrl(docCommatrixVersionedUrl, fp)
+		docComMatrixCreator, err := commatrixcreator.New(epExporter, fp, types.FormatCSV, env, deployment)
+		o.Expect(err).ToNot(o.HaveOccurred())
+		docComDetailsList, err := docComMatrixCreator.GetComDetailsListFromFile()
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		g.By("Filter documented commatrix for diff generation")
+		// if cluster is a SNO exclude standard deployment type's static entries in diff generation
+		if isSNO {
+			// Exclude static entries of standard deployment type and all worker nodes entries.
+			docComDetailsList = excludeGivenStaticEntries(docComDetailsList, &types.ComMatrix{Matrix: types.StandardStaticEntries}, &types.ComMatrix{Matrix: docComDetailsList})
+		} else {
+			// Exclude specific master entries that are documented as both worker and master,
```

Comment:
Maybe the comment next to the StandardExcludedMasterComDetails var is enough and here we can just mention in general we exclude from standard (in case we add more due to different reasons)